crates/holochain/src/core/workflow/app_validation_workflow.rs

```diff
@@ -312,7 +312,10 @@ pub fn op_to_record(op: Op, omitted_entry: Option<Entry>) -> Record {
     match op {
         Op::StoreRecord(StoreRecord { mut record }) => {
             if let Some(e) = omitted_entry {
-                *record.as_entry_mut() = RecordEntry::Present(e);
+                // NOTE: this is only possible in this situation because we already removed
+                // this exact entry from this Record earlier. DON'T set entries on records
+                // anywhere else without recomputing hashes and signatures!
+                record.entry = RecordEntry::Present(e);
```

Comment:
Doesn't have to be a problem for this PR, but is there a unit test verifying this behavior? I'm sure this is hit by functional tests and it's existing behavior.

If not can this be tracked somewhere to come back to?

Reply:
Not sure what you mean. Do you mean testing what happens when somebody actually does this unwittingly, i.e. the record Entry doesn't match the Action? It will certainly get caught in validation, either on your own node or someone else's (the former being bad, the latter being catastrophically bad), and may lead to UB along the way. I think it would be onerous to check the validity of a Record in every place we use one though.

The thing that would be useful to test is that nobody ever does this, but that would be a lint and not a test.